client/src/main/java/tds/assessment/Assessment.java

```diff
@@ -12,9 +12,18 @@
     private String selectionAlgorithm;
     private float startAbility;
     private String subject;
+    private List<ItemConstraint> itemConstraints;
     private List<Segment> segments = new ArrayList<>();
 
-    private Assessment() {}
+    public Assessment() {}
```

Comment:
Todd, I think that's a great point. There are some constructors (like segment or item) that take in their respective keys as constructor arguments - I think if it makes sense, we should add as many required fields to the constructor as possible. At what point do we decide there are too many arguments to include in the constructor, though?

Regarding setters - I'd be fine with removing the setters that aren't needed. Curious to hear other people's thoughts on this.